packages/core/src/BaseServiceClient.js

```diff
@@ -59,15 +59,22 @@ class BaseServiceClient {
    * @returns {Promise<ChannelStateReply>}
    */
   async getChannelState(channelId) {
-    const signatureBytes = await this._account.signData({ t: 'uint256', v: channelId });
+    const currentBlockNumber = await this._web3.eth.getBlockNumber();
+    const channelIdStr = new BigNumber(channelId).toFixed();
+    const signatureBytes = await this._account.signData(
+      { t: 'string', v: '__get_channel_state' },
+      { t: 'address', v: this._mpeContract.address },
+      { t: 'uint256', v: channelIdStr },
+      { t: 'uint256', v: currentBlockNumber },
+    );
 
     const channelIdBytes = Buffer.alloc(4);
     channelIdBytes.writeUInt32BE(channelId, 0);
 
-    const ChannelStateRequest = this._getChannelStateRequestMethodDescriptor();
-    const channelStateRequest = new ChannelStateRequest();
+    const channelStateRequest = new this.paymentChannelStateServiceClient.getChannelState.requestType();
```

Comment:
Need to revert this as this does not work for web platform.

Reply:
Addressed in [commit](https://github.com/singnet/snet-sdk-js/pull/17/commits/41d3d70562fbb29327792853bd628672ecaf0aa9)